Sources/Sentry/UIViewController+Sentry.m

```diff
@@ -1,32 +0,0 @@
-#import "UIViewController+Sentry.h"
-
-#if SENTRY_HAS_UIKIT
-
-@implementation
-UIViewController (Sentry)
-
-- (NSArray<UIViewController *> *)sentry_descendantViewControllers
```

Comment:
`h`: I would rather keep the file and its tests, please.

Reply:
That is fine, I can do that. I removed the category because it didn't work with my first attempt, which was to conditionally define macros for every UIKit thing. Now that I can just gate compilation of UIKit things, I can do it here too 👍🏻